src/main/java/org/jabref/logic/xmp/XMPUtil.java

```diff
@@ -318,51 +297,106 @@ public static PDDocument loadWithAutomaticDecryption(File file) throws IOExcepti
                 }
             }
         }
+    }
 
-        /*
-         * Rights -> Rights
-         */
-        s = dcSchema.getRights();
+    private static void setRights(BibEntry entry, DublinCoreSchema dcSchema) {
+        String s = dcSchema.getRights();
         if (s != null) {
             entry.setField("rights", s);
         }
+    }
 
-        /*
-         * Source -> Source
-         */
-        s = dcSchema.getSource();
+    private static void setSource(BibEntry entry, DublinCoreSchema dcSchema) {
+        String s = dcSchema.getSource();
         if (s != null) {
             entry.setField("source", s);
         }
+    }
 
-        /*
-         * Subject -> Keywords
-         */
+    private static void setSubject(BibEntry entry, DublinCoreSchema dcSchema,
+            Character keyWordSeparator) {
         List<String> subjects = dcSchema.getSubjects();
         if (subjects != null) {
-            entry.addKeywords(subjects, xmpPreferences.getKeywordSeparator());
+            entry.addKeywords(subjects, keyWordSeparator);
         }
+    }
 
-        /*
-         * Title -> Title
-         */
-        s = dcSchema.getTitle();
+    private static void setTitle(BibEntry entry, DublinCoreSchema dcSchema) {
+        String s = dcSchema.getTitle();
         if (s != null) {
             entry.setField(FieldName.TITLE, s);
         }
+    }
 
-        /*
-         * Type -> Type
-         */
+    private static void setType(BibEntry entry, DublinCoreSchema dcSchema) {
         List<String> l = dcSchema.getTypes();
         if ((l != null) && !l.isEmpty()) {
-            s = l.get(0);
+            String s = l.get(0);
             if (s != null) {
                 entry.setType(s);
             }
         }
+    }
+
+    /**
+     * Helper function for retrieving a BibEntry from the DublinCore metadata
+     * in a PDF file.
+     *
+     * To understand how to get hold of a DublinCore have a look in the
+     * test cases for XMPUtil.
+     *
+     * The BibEntry is build by mapping individual fields in the dublin core
+     * (like creator, title, subject) to fields in a bibtex entry.
+     *
+     * @param dcSchema The document information from which to build a BibEntry.
+     * @return The bibtex entry found in the document information.
+     */
+    public static Optional<BibEntry> getBibtexEntry(DublinCoreSchema dcSchema,
+            XMPPreferences xmpPreferences) {
+
+        BibEntry entry = new BibEntry();
+
+        // Contributor -> Editor
```

Comment:
I would not comment these functions as the implementation might change... The comments could co at `setEditor` etc. as method comment.